### PR TITLE
Add more tests around service mesh enabling

### PR DESCRIPTION
**Pull request by rgulewich**
As a follow-on to the last commit.

### CODE REVIEW

File: executor/runtime/types/container_test.go [hunk 1]
```diff
@@ -668,3 +668,41 @@ func TestServiceMeshEnabled(t *testing.T) {
 	assert.NotNil(t, svcMeshConf)
 	assert.Equal(t, svcMeshConf.Image, imgName)
 }
+
+func TestServiceMeshEnabledWithConfig(t *testing.T) {
+	// If service mesh is set to enabled, but neither the `ProxydServiceImage` config value
+	// or the passhtrough property are set, service mesh should end up disabled
+	config := config.Config{
+		ContainerServiceMeshEnabled: true,
```
Comment (solarkennedy) [inline executor/runtime/types/container_test.go:676]: Can you add a negative/absent test, where we have an empty config and we assert that there is not a servicemesh sidecar?

Comment (rgulewich) [reply executor/runtime/types/container_test.go:676]: Yeah, I added that in the last commit.  It's part of `TestNewContainer()` above.

**Review by solarkennedy — APPROVED** [review]

**Review by alekar — APPROVED** [review]
